packages/NodeNameResolver/NodeNameResolver.php

```diff
@@ -181,7 +181,15 @@ public function isStringName(string $resolvedName, string $desiredName): bool
             return $desiredName === $resolvedName;
         }
 
-        return strcasecmp($resolvedName, $desiredName) === 0;
+        if (strcasecmp($resolvedName, $desiredName) === 0) {
+            return true;
+        }
+
+        if (str_contains($desiredName, '*')) {
+            return fnmatch($desiredName, $resolvedName, FNM_NOESCAPE);
+        }
+
+        return false;
```

Comment:
@TomasVotruba @staabm I am thinking that we can enable again regex support to avoid BC break, with check string first char regex delimiter or contains *:

```suggestion
        if ($this->regexPatternDetector->isRegexPattern($desiredName)) {
            return StringUtils::isMatch($resolvedName, $desiredName);
        }

        if (str_contains($desiredName, '*')) {
            return fnmatch($desiredName, $resolvedName, FNM_NOESCAPE);
        }

        return false;
```

The check is always check with string comparison first, so it should be fast enough, wdyt?

Reply:
I am fine for BC reasons. But for 1.0 it should still be dropped IMO

Reply:
The fnmatch removal support was cause a lot of errors when enabling `rules-tests` at phpunit.xml on PR:

- https://github.com/rectorphp/rector-phpunit/pull/251#pullrequestreview-1618707993

with patched with `fnmatch()` everywhere.

Since it check string with `in_array() internally` and `str_contains()` and  check early before executing regex, I think it fast enough

Reply:
I wonder why this error did not show up in the PRs directly?

Reply:
Yes, the `rules-tests` was missing registered in `rector-phpunit`'s phpunit.xml, by register `rules-tests` in phpunit.xml, the error show up, I fixed that by apply `fnmatch()` everywhere :)

see https://github.com/rectorphp/rector-phpunit/pull/251/files#diff-0469c1256426c98149a652ef838fd62475a18e529dc723ca0e8faa6c3fd8749e

The problem I can see is the user that directly use it will see BC break a lot and probably don't know what to do if the changes are a lot.

Reply:
I added back regex support, with check string start with regex delimiter https://github.com/rectorphp/rector-src/pull/4964/commits/209afb2ee7d70525d2155c1df05550c980c837f2